packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py

```diff
@@ -2130,29 +2216,49 @@ def _prepare_job_params(self):
             ]
             random.shuffle(label_data_roi_ids)
 
-            for job_data_roi_ids in take_by(
-                label_data_roi_ids, int(job_size_mult * self.manifest.annotation.job_size)
-            ):
-                job_gt_count = max(
-                    self.manifest.validation.val_size, int(gt_ratio * len(job_data_roi_ids))
-                )
-                job_gt_count = min(len(label_gt_roi_ids), job_gt_count)
-
-                job_gt_roi_ids = random.sample(label_gt_roi_ids, k=job_gt_count)
-
-                job_roi_ids = list(job_data_roi_ids) + list(job_gt_roi_ids)
-                random.shuffle(job_roi_ids)
-
-                job_params.append(self._JobParams(label_id=label_id, roi_ids=job_roi_ids))
+            task_params.extend(
+                [
+                    self._TaskParams(
+                        label_id=label_id, roi_ids=task_data_roi_ids, roi_gt_ids=label_gt_roi_ids
+                    )
+                    for task_data_roi_ids in take_by(
+                        label_data_roi_ids, Config.cvat_config.cvat_max_jobs_per_task * segment_size
+                    )
+                ]
+            )
 
-        self._job_params = job_params
+        self._task_params = task_params
 
     def _prepare_job_labels(self):
         self.point_labels = {}
 
         for skeleton_label in self.manifest.annotation.labels:
             for point_name in skeleton_label.nodes:
-                self.point_labels[(skeleton_label.name, point_name)] = point_name
+                self.point_labels[(skeleton_label.name, point_name)] = point_name  # ??
+
+    def _prepare_points_mapping(self):
+        # (skeleton_label, point_label) to (dataset_item.id, point)
+        self.points = {}
+
+        # GT should contain all GT images with only one point per skeleton node
+        for dataset_item in self._gt_dataset:
+            for skeleton in dataset_item.annotations:
+                if not isinstance(skeleton, dm.Skeleton):
+                    continue
+
+                for point in skeleton.elements:
+                    point_label_name = (
+                        self._gt_dataset.categories()[dm.AnnotationType.label]
+                        .items[point.label]
+                        .name
+                    )
+
+                    # TODO: exclude such images
+                    if dm.Points.Visibility.absent in point.visibility:
+                        continue
```

Comment:
Why such images should be excluded?

Reply:
Is there an option to skip frames that "should not be annotated" in CVAT `single shape mode`?

Reply:
I just thought if we know from the GT that some frames contain outside points, why should we include such images in the dataset to be annotated? Why should we mislead an annotator?

Reply:
Yes, there is a skip button:

![image](https://github.com/user-attachments/assets/9eba7644-f2a3-4f91-bc1d-b141cadf84aa)

The image should be skipped in such situations.

Reply:
Now points with `dm.Points.Visibility.absent` or `dm.Points.Visibility.hidden` visibilities are excluded from the GT annotations uploaded to CVAT GT jobs. Related frames are not excluded from the annotation dataset.